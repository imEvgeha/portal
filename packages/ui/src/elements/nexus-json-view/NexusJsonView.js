import React from 'react';
import PropTypes from 'prop-types';
import ReactJsonView from 'react-json-view';
import NexusResizable from '../nexus-resizable/NexusResizable';
import './NexusJsonView.scss';

const NexusJsonView = ({defaultHeight, defaultWidth, ...rest}) => {
    return (
        <NexusResizable defaultHeight={defaultHeight} defaultWidth={defaultWidth}>
            <div className="nexus-c-json-view">
                <ReactJsonView
                    name={false}
                    enableClipboard={false}
                    displayObjectSize={false}
                    displayPropTypes={false}
                    {...rest}
                />
            </div>
        </NexusResizable>
    );
};

NexusJsonView.propTypes = {
    defaultHeight: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    defaultWidth: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
};

NexusJsonView.defaultProps = {
    defaultHeight: 250,
    defaultWidth: '100%',
};

export default NexusJsonView;
