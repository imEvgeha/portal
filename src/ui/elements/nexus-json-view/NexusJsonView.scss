.nexus-c-json-view {
    padding: 20px;
    background-color: #fff;
    border: 1px solid #bdc3c7;
    overflow: auto;
    height: 100%;
}
