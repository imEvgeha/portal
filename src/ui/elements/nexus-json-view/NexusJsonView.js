import React from 'react';
import ReactJsonView from 'react-json-view';
import {Resizable} from 're-resizable';
import './NexusJsonView.scss';

const NexusJsonView = props => (
    <Resizable
        defaultSize={{
            height: 250
        }}
        minWidth='100%'
        maxWidth='100%'
        enable={{top:false, right:false, bottom:true, left:false, topRight:false, bottomRight:true, bottomLeft:false, topLeft:false}}
    >
        <div className="nexus-c-json-view">
            <ReactJsonView
                name={false}
                enableClipboard={false}
                displayObjectSize={false}
                displayPropTypes={false}
                {...props}
            />
        </div>
    </Resizable>
);

export default NexusJsonView;
