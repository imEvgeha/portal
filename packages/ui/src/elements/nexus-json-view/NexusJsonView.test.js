import React from 'react';
import {shallow} from 'enzyme';
import NexusJsonView from './NexusJsonView';

describe('NexusJsonView', () => {
    let wrapper = null;

    beforeEach(() => {
        const props = {
            src: {},
        };
        wrapper = shallow(<NexusJsonView {...props} />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with default height and width', () => {
        const nexusResizableElement = wrapper.find('NexusResizable');
        expect(nexusResizableElement.html()).toContain('width:100%');
        expect(nexusResizableElement.html()).toContain('height:250px');
    });

    it('should render with specified height and width', () => {
        wrapper.setProps({
            defaultWidth: 300,
            defaultHeight: 700,
        });
        wrapper.update();
        const nexusResizableElement = wrapper.find('NexusResizable');
        expect(nexusResizableElement.html()).toContain('width:300px');
        expect(nexusResizableElement.html()).toContain('height:700px');
    });
});
