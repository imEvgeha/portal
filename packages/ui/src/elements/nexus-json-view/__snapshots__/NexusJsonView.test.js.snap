// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NexusJsonView should match snapshot 1`] = `
<NexusResizable
  defaultHeight={250}
  defaultWidth="100%"
  handles={
    Object {
      "bottom": true,
    }
  }
>
  <div
    className="nexus-c-json-view"
  >
    <n
      collapseStringsAfterLength={false}
      collapsed={false}
      defaultValue={null}
      displayArrayKey={true}
      displayDataTypes={true}
      displayObjectSize={false}
      displayPropTypes={false}
      enableClipboard={false}
      groupArraysAfterLength={100}
      iconStyle="triangle"
      indentWidth={4}
      name={false}
      onAdd={false}
      onDelete={false}
      onEdit={false}
      onSelect={false}
      quotesOnKeys={true}
      shouldCollapse={false}
      sortKeys={false}
      src={Object {}}
      style={Object {}}
      theme="rjv-default"
      validationMessage="Validation Error"
    />
  </div>
</NexusResizable>
`;
